com.unity.multiplayer.mlapi/Tests/Runtime/SceneLoadingTest.cs

```diff
@@ -0,0 +1,140 @@
+using System.Collections;
+using System.IO;
+using System.Reflection;
+using UnityEngine;
+using NUnit.Framework;
+using UnityEngine.TestTools;
+using UnityEngine.SceneManagement;
+using UnityEditor.SceneManagement;
+
+namespace MLAPI.RuntimeTests
+{
+    public class SceneLoadingTest
+    {
+        private NetworkManager m_NetworkManager;
+
+        private bool m_SceneLoaded;
+        private string m_TargetSceneNameToLoad;
+        private Scene m_LoadedScene;
+
+        [UnityTest]
+        public IEnumerator SceneLoading()
+        {
+            // Load the first scene with the predefined NetworkManager
+            var execAssembly = Assembly.GetExecutingAssembly();
+            var packagePath = UnityEditor.PackageManager.PackageInfo.FindForAssembly(execAssembly).assetPath;
+            var scenePath = Path.Combine(packagePath, "Tests/Runtime/TestScenes/SceneLoadingTest.unity");           
+            m_TargetSceneNameToLoad = scenePath;
+            SceneManager.sceneLoaded += SceneManager_sceneLoaded;
+            EditorSceneManager.LoadSceneAsyncInPlayMode(scenePath, new LoadSceneParameters(LoadSceneMode.Additive));
+
+            //Wait for the scene to load
+            var timeOut = Time.realtimeSinceStartup + 2;
+            var timedOut = false;
+            while(!m_SceneLoaded)
+            {
+                yield return new WaitForSeconds(0.01f);
+                if(timeOut < Time.realtimeSinceStartup)
+                {
+                    timedOut = true;
+                    break;
+                }
+            }
+
+            Assert.IsFalse(timedOut);
+            Assert.IsTrue(m_SceneLoaded);
+            Assert.NotNull(m_LoadedScene);
+
+            // Set the scene to be active if it is not the active scene
+            if (SceneManager.GetActiveScene().name != m_LoadedScene.name)
+            {
+                Debug.Log($"Loaded scene not active, activating scene: {scenePath}");
+                SceneManager.SetActiveScene(m_LoadedScene);
+                Assert.IsTrue(SceneManager.GetActiveScene().name == m_LoadedScene.name);
+            }
+
+            // Get the NetworkManager instantiated from the scene
+            var gameObject = GameObject.Find("NetworkManager");
+
+            Assert.IsNotNull(gameObject);
+
+            m_NetworkManager = gameObject.GetComponent<NetworkManager>();
```

Comment:
I guess what I wonder here is, are we discovering here that in fact we want to have a way in general, not just for this test, for the library to wait to start communications until an async-loaded scene is loaded?

Reply:
Up until this experiment, we were constructing our own instances of NetworkManager without having a pre-defined scene that already had a NetworkManager with configuration settings specific to the unit test in question.  What this provides is the ability to **both** load a scene as the "basic NetworkManager configuration" and then also provide the capability to do scene transitions during runtime via MLAPI's NetworkSceneManager.SwitchScene method.

The reason I am waiting until the scene is done loading is that I want to make sure the NetworkManager in the default/first scene is fully loaded before trying to find my NetworkManager instance that was instantiated during the loading of the scene.